catboost/private/libs/target/calc_class_weights.cpp

```diff
@@ -0,0 +1,42 @@
+#include "calc_class_weights.h"
+
+#include <catboost/libs/logging/logging.h>
+
+#include <util/generic/algorithm.h>
+#include <util/system/yassert.h>
+
+static std::function<float(ui64, ui64)> GetWeightFunction(EAutoClassWeightsType autoClassWeightsType) {
+    switch (autoClassWeightsType) {
+        case EAutoClassWeightsType::Balanced:
+            return [](ui64 maxClassSize, ui64 classSize) -> float {
+                return classSize > 0 ? static_cast<float>(maxClassSize) / classSize : 1.f;
+            };
+        case EAutoClassWeightsType::SqrtBalanced:
+            return [](ui64 maxClassSize, ui64 classSize) -> float {
+                return classSize > 0 ? sqrt(static_cast<float>(maxClassSize) / classSize) : 1.f;
+            };
+        case EAutoClassWeightsType::None:
+            Y_VERIFY(false);
+    }
+}
+
+TVector<float> CalculateClassWeights(TConstArrayRef<float> targetClasses, ui32 classCount, EAutoClassWeightsType autoClassWeightsType) {
+    Y_VERIFY(classCount > 0);
+    TVector<ui64> numClassItems(classCount, 0ll);
+
+    for (const auto& target : targetClasses) {
```

Comment:
Let's make it parallel with LocalExecutor

Reply:
done